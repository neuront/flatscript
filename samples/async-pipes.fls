list: [4, 2, 3, 1]

func print(element, cb)
    setTimeout(%, 1)
    console.log('Element: ' + element)
    cb()

func echo(cb, index)
    setTimeout(%, 1)
    console.log('Index: ' + index)
    cb()

func g(p, q, cb)
    setTimeout(cb, 1)
    return 0

x: list |: g(print($, %), echo(%, $i), %)
console.log(x)

func asyncSqr(element, cb)
    setTimeout(%, 1)
    cb(element * element)

console.log(list |:
    asyncSqr($, %x)
    $result.push(x)
)

console.log(list |:
    asyncSqr($, %x)
    if x % 3 = 1
        $r.push(x)
)

list |:
    console.log($i + ':' + $)
    setTimeout(%, 1)

console.log('-- references in nested --')

func nestedPiping(x, %%)
    x |:
        i: $
        x |:
            setTimeout(%, 1)
            console.log('+', i + $)
        console.log('.', $, $i)

nestedPiping([0, 1], %%)
nestedPiping('ab', %%)

console.log('End')
