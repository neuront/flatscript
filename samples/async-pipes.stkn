list: [4, 2, 3, 1]

func print(element, cb)
    setTimeout(%, 1)
    console.log('Element: ' + element)
    cb()

func echo(cb, index)
    setTimeout(%, 1)
    console.log('Index: ' + index)
    cb()

func g(p, q, cb)
    setTimeout(cb, 1)
    return 0

x: list |: g(print($, %), echo(%, $i), %)
console.log(x)
