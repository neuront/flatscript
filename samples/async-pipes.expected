Element: 4
Index: 0
Element: 2
Index: 1
Element: 3
Index: 2
Element: 1
Index: 3
[ 0, 0, 0, 0 ]
[ 16, 4, 9, 1 ]
[ 16, 4, 1 ]
0:4
1:2
2:3
3:1
End
